packages/kafka-iam-auth/src/index.ts

```diff
@@ -364,7 +421,26 @@ export const runConsumer = async (
   consumerHandler: (messagePayload: EachMessagePayload) => Promise<void>
 ): Promise<void> => {
   try {
-    await initConsumer(config, topics, consumerHandler);
+    await initCustomConsumer({ config, topics, consumerHandler });
+  } catch (e) {
+    genericLogger.error(
+      `Generic error occurs during consumer initialization: ${e}`
+    );
+    processExit();
+  }
+};
+
+export const runBatchConsumer = async (
```

Comment:
this would be the refactor with the previous suggestion

```
export const runBatchConsumer = async (
  config: KafkaBatchConsumerConfig,
  topics: string[],
  consumerHandlerBatch: (messagePayload: EachBatchPayload) => Promise<void>
): Promise<void> => {
  try {
    const consumerRunConfig = (consumer: Consumer) => ({
      autoCommit: false,
      eachBatch: async (payload: EachBatchPayload) => {
        try {
          await consumerHandlerBatch(payload);
          await kafkaCommitBatchOffsets(consumer, payload);
        } catch (e) {
          throw kafkaMessageProcessError(
            payload.batch.topic,
            payload.batch.partition,
            payload.batch.lastOffset.toString(),
            e
          );
        }
      },
    });
    await initCustomConsumer({
      config,
      topics,
      consumerRunConfig,
      batchConfig: { 
        minBytes: config.averageKafkaMessageSizeInBytes * config.messagesToReadPerBatch,
        maxWaitTimeInMs: config.maxWaitKafkaBatch,
    }
    });
  } catch (e) {
    genericLogger.error(
      `Generic error occurs during consumer initialization: ${e}`
    );
    processExit();
  }
};
```